featurebyte/query_graph/graph.py

```diff
@@ -263,7 +265,13 @@ def extract_table_id_feature_job_settings(
         node_table_id_iterator = self.iterate_group_by_node_and_table_id_pairs(
             target_node=target_node
         )
+        table_ids = set()
         for group_by_node, table_id in node_table_id_iterator:
+            if keep_first_only and table_id in table_ids:
+                # skip duplicate table
```

Comment:
In which cases will we want to keep duplicate tables?

Reply:
When we validate whether the duplicate tables have the same setting or not in https://github.com/featurebyte/featurebyte/blob/main/featurebyte/service/feature.py#L117-L148.

Reply:
🤔 Maybe we should make the `keep_first_only` default to `True` & update the parameter there.

Reply:
I think either default value is fine. Would be helpful to have a comment on this.